src/rvs_blas.cpp

```diff
@@ -758,91 +797,129 @@ void rvs_blas::generate_random_matrix_data(void) {
 
       for (i = 0; i < size_c; ++i)
-        ((struct rocblas_f8* )hdc)[i] = rocblas_f8(fast_pseudo_rand(&nextr));
+        ((struct rocblas_f8* )hdc)[i] = rocblas_f8(fast_pseudo_rand(&nextr, i));
     }
 
     // 16-bit floating point real (fp16_r) format
     if(data_type == "fp16_r") {
 
       for (i = 0; i < size_a; ++i)
-        ((rocblas_half* )hda)[i] = rocblas_half(fast_pseudo_rand(&nextr));
+        ((rocblas_half* )hda)[i] = rocblas_half(fast_pseudo_rand(&nextr, i));
 
       for (i = 0; i < size_b; ++i)
-        ((rocblas_half* )hdb)[i] = rocblas_half(fast_pseudo_rand(&nextr));
+        ((rocblas_half* )hdb)[i] = rocblas_half(fast_pseudo_rand(&nextr, i));
 
       for (i = 0; i < size_c; ++i)
-        ((rocblas_half* )hdc)[i] = rocblas_half(fast_pseudo_rand(&nextr));
+        ((rocblas_half* )hdc)[i] = rocblas_half(fast_pseudo_rand(&nextr, i));
     }
 
     // 16-bit brain floating point real (bp16_r) format
     if(data_type == "bp16_r") {
 
       for (i = 0; i < size_a; ++i)
-        ((struct rocblas_bfloat16* )hda)[i] = rocblas_bfloat16(fast_pseudo_rand(&nextr));
+        ((struct rocblas_bfloat16* )hda)[i] = rocblas_bfloat16(fast_pseudo_rand(&nextr, i));
 
       for (i = 0; i < size_b; ++i)
-        ((struct rocblas_bfloat16* )hdb)[i] = rocblas_bfloat16(fast_pseudo_rand(&nextr));
+        ((struct rocblas_bfloat16* )hdb)[i] = rocblas_bfloat16(fast_pseudo_rand(&nextr, i));
 
       for (i = 0; i < size_c; ++i)
-        ((struct rocblas_bfloat16* )hdc)[i] = rocblas_bfloat16(fast_pseudo_rand(&nextr));
+        ((struct rocblas_bfloat16* )hdc)[i] = rocblas_bfloat16(fast_pseudo_rand(&nextr, i));
     }
+  }
+}
 
+float rvsblas_uniform_int_1_10()
+{
+  if(!rvsblas_t_rand_init)
+  {
+    for(int i = 0; i < RANDBUF; i++)
+    {
+      rvsblas_t_rand_f_array[i]
+        = (float)std::uniform_int_distribution<unsigned>(1, 10)(rvsblas_t_rng);
+      rvsblas_t_rand_d_array[i] = (double)rvsblas_t_rand_f_array[i];
+    }
+    rvsblas_t_rand_init = 1;
   }
+  rvsblas_t_rand_idx = (rvsblas_t_rand_idx + 1) & (RANDLEN - 1);
+  return rvsblas_t_rand_f_array[rvsblas_t_rand_idx];
 }
 
 /**
  * @brief fast pseudo random generator 
  * @return floating point random number
  */
-float rvs_blas::fast_pseudo_rand(uint64_t *nextr) {
+float rvs_blas::fast_pseudo_rand(uint64_t *nextr, size_t i) {
+
+  if ("rand" == matrix_init) {
+
+    if("fp8_r" == data_type) {
+      return (float)std::uniform_int_distribution<int>(1, 2)(rvsblas_t_rng);
+    }
+    else if (("fp16_r" == data_type) || ("hgemm" == ops_type))
+    {
+      return (float)std::uniform_int_distribution<int>(-2, 2)(rvsblas_t_rng);
+    }
+    else if("bf16_r" == data_type)
+    {
+      return (float)std::uniform_int_distribution<int>(-2, 2)(rvsblas_t_rng);
+    }
+    else { /* sgemm, dgemm */
+      return rvsblas_uniform_int_1_10();
+    }
+  }
+  else if ("trig" == matrix_init) {
+    return static_cast<float>(sin(i));
```

Comment:
better to cast argument to float than to cast return type? looks like arg, if int, will be treated as double which returns double as result and we are downcasting it to float in this case?

Reply:
Done